src/Frontend/Core/Engine/Base/Block.php

```diff
@@ -85,6 +87,20 @@ class Block extends Object
     private $templatePath;
 
     /**
+     * TwigTemplate instance
+     *
+     * @var TwigTemplate
+     */
+    protected $tpl;
+
+    /**
+     * URL instance
+     *
+     * @var Url
+     */
+    protected $URL;
```

Comment:
Shouldn't this be `protected $url`, in the past we used uppercase for abbreviation, but this conflicts with using uppercase for constants.